drivers/wifi/nrf700x/src/fmac_main.c

```diff
@@ -324,12 +338,33 @@ int nrf_wifi_reg_domain(const struct device *dev, struct wifi_reg_domain *reg_do
 			goto err;
 		}
 	} else if (reg_domain->oper == WIFI_MGMT_GET) {
+
+		if (!reg_domain->chan_info)	{
+			LOG_ERR("%s: No memory allocated for regulatory info\n", __func__);
+			goto err;
+		}
+
 		status = nrf_wifi_fmac_get_reg(rpu_ctx_zep->rpu_ctx, &reg_domain_info);
 		if (status != NRF_WIFI_STATUS_SUCCESS) {
 			LOG_ERR("%s: Failed to get regulatory domain", __func__);
 			goto err;
 		}
 		memcpy(reg_domain->country_code, reg_domain_info.alpha2, WIFI_COUNTRY_CODE_LEN);
+
+		reg_domain->num_channels = reg_domain_info.reg_chan_count;
+		for (chan_idx = 0; chan_idx < reg_domain_info.reg_chan_count; chan_idx++) {
+			chan_info = &(reg_domain->chan_info[chan_idx]);
+			chan_info->center_frequency =
+				reg_domain_info.reg_chan_info[chan_idx].center_frequency;
```

Comment:
`reg_domain_info.reg_chan_info[chan_idx]` storing in an intermediate variable will solve the indent issue.